pylint/lint/pylinter.py

```diff
@@ -851,6 +852,7 @@ def _get_file_descr_from_stdin(self, filepath: str) -> Iterator[FileItem]:
             self.config.ignore_patterns,
             self.config.ignore_paths,
         ):
+            self.skipped_paths.add(filepath)
```

Comment:
Can we instead store this as a an attribute on `LinterStats`? Preferably just as an `int` as we never actually need the file name. That would centralise all statistics counting.

Reply:
I had that originally! I had two concerns that made me switch
* the ignore logic happens at a few different layers. From the way it's structured, there should be no overlap, but I was weary of double-counting
* in the next step (or potentially in this PR itself, depending on how you feel) I'd also like to use `get_module_files` to get an accurate file count rather than a “file or module" count; it would also enable pylint to report precisely what got skipped, though I guess that could be reported as we iterate and ignore them

If you're cool with the first, I can switch it back and leave the second point to a future discussion.